### DISCUSSION TITLE
Disable log messages about reconcilation

### DISCUSSION THREAD

**Discussion by candrii** [fluxcd/flux2#2204] [General]
There are too much log messages from fluxcd components like :

``` 
11:24:10.565
reconciliation finished in 59.360168ms, next run in 30s
 
11:24:11.243
Reconciliation finished in 3.479882262s, next run in 1m0s
 
11:24:12.025
reconciliation finished in 60.137738ms, next run in 30s
 
11:24:12.035
Discarding event, no alerts found for the involved object
 
11:24:12.039
Discarding event, no alerts found for the involved object
 
11:24:12.040
Discarding event, no alerts found for the involved object
 
11:24:13.262
Reconciliation finished in 284.119914ms, next run in 1m0s
 
11:24:13.492
Reconciliation finished in 3.185145336s, next run in 1m0s
```

Are there some way to disable that log messages from source-controller?

**Comment by kingdonb** [comment]
There is a change related to this in rewrite-dev of source-controller

We are on the fence - Kubernetes log does not tell you about reconciliations that had no effect. But those take place inside of the cluster and in full control of the Kubernetes cluster. (If there was a write to be undone, the change that propagated that drift would show up in the log as well.) Flux's source is external, and those events without any change contained in them could be failing for many reasons outside of Flux's control. We thought they should be logged.

There are logging changes and eventing changes coming to all Flux controllers, and one of the major goals is simplifying the output so that there is less spam. Many advancements have been made already, in the past few months. This may not have changed yet though.

In general, the logs are intended to be more verbose, and Events are used for "when something actually happened." You can monitor events in a Slack channel or other means through notification-controller Providers. The changes we're making are around the notion that events are from a point in time, and users don't want to trace through some historical event log to find out what happened, they just want to know what state it's in now, and why it's stalled if it has stalled. Those ideas can be better represented as Conditions in the status than events. So, soon Flux will update the APIs to include more details in the Status.Condition.

Back to this issue, we have had some bugs where "everything stopped" and there was no information in the logs about why it stopped, so it is important to log when a reconciliation happened, even if it did not change anything. In my opinion this could be an "INFO" message, you should be able to configure the flux log levels for "WARN" only, so you see exceptional events only, and messages about ordinary goings-on which are not errors or trouble indicators will be missing from the logs.

But if we go down that road, then it's also likely you and I will not agree on where to draw the line. (Do we want to see logged when a change was applied successfully without error? ... what if a new commit applied had no changes? Any of these could be logged info.)

In some projects, they use verbosity numbers for more granularity. I am not a fan of that approach for Flux, as it tends toward a "knob" and I'm warming up to the "no knobs" camp.

If you are using Events and notification controller, then you can filter at the `Alert` level by selecting the resources that you're interested in watching for events, and filter through `spec.filter` any messages that are generally considered un-interesting.

For example, I have filtered out all "dependency is not ready" messages, because I have added Prometheus AlertManager and it can tell me when a dependency is not ready for longer than (x) minutes, in the form of an alert from AlertManager. (I wish I didn't have to do that, but I have a lot of dependencies between my Kustomizations and the dependson waiting message is noisy and generally not very helpful for me.)
